forem/forem
# [Team Email Login] Refactor the Email Authentication flow

## What type of PR is this? (check all applicable)


## Description
Per [design specs](https://user-images.githubusercontent.com/816402/96673429-58375c80-131b-11eb-848a-0ccacec0e0cf.jpg), this PR refactors the process of enabling/disabling Email-Password Authentication for Forems.

## Related Issues
- https://github.com/forem/InternalProjectPlanning/issues/166
- https://github.com/forem/InternalProjectPlanning/issues/169

## QA Instructions, Screenshots, Recordings
🎥  **A video demo appears below these instructions**

1. Navigate to `localhost:xxxx/admin/config`. Ensure that your user account has the `single_resource_admin` role. To add this role, run `User.find(<your-user-id>).add_role("single_resource_admin").save!` in your rails console.
2. Toggle open the `Authentication` section. Click the `Enable` or `Edit` button to the right of the `Email address` label (the button's text depends on whether email auth has been enabled for the forem). The button should disappear, and a section containing the reCAPTCHA fields, a `Disable` button and a `Cancel` button should appear below.
3. Run through the various flows for enabling/disabling email authentication. A few things to verify:
- That disabling email authentication also turns Invite-Only Mode off (its checkbox should be unchecked).
- That disabling email authentication removes the option of email sign in at `localhost:xxxx/enter` and removes the option of email registration at `localhost:xxxx/enter?state=new-user`.
- That the Invite-Only Mode checkbox is disabled when Email Auth is turned off.
- That the option to turn Email Auth off is absent when Invite-Only Mode is enabled.
---

https://www.loom.com/share/666d3ca2832045d5a3b409d57e2a7235


## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?
No

## [optional] What gif best describes this PR or how it makes you feel?

![Dancing Pumpkin](https://media.giphy.com/media/xThtakc1HIEmhQgBhK/giphy.gif)
